### PR TITLE
Add new feature: Dns cheat

### PR DESCRIPTION
**Pull request by arcanique**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind feature

**What this PR does / why we need it**:
>
> this PR implement dns cheat for edgemesh to resolve domain of the service

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: edgemesh/pkg/server/dns.go [hunk 1]
```diff
@@ -0,0 +1,401 @@
+package server
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"github.com/golang/glog"
+
+	//"github.com/golang/glog"
```
Comment (sids-b) [inline edgemesh/pkg/server/dns.go]: please remove commented import and the blank line above

File: edgemesh/pkg/server/dns.go [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+	"errors"
+	"net"
+	"os"
+	"strings"
+	"unsafe"
+)
+
+var (
+	inter = "以太网 3"
```
Comment (sids-b) [inline edgemesh/pkg/server/dns.go]: English please

File: edgemesh/pkg/server/dns.go [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	idxadd := unsafe.Sizeof(h.additNum) + idxaut
+
+	binary.BigEndian.PutUint16(rspHead[: idxTran], h.transactionID)
+	binary.BigEndian.PutUint16(rspHead[idxTran: idxflag], h.flags)
+	binary.BigEndian.PutUint16(rspHead[idxflag: idxque], h.queNum)
+	binary.BigEndian.PutUint16(rspHead[idxque: idxans], h.ansNum)
+	binary.BigEndian.PutUint16(rspHead[idxans: idxaut], h.authNum)
+	binary.BigEndian.PutUint16(rspHead[idxaut: idxadd], h.additNum)
+	return
+}
```
Comment (sids-b) [inline edgemesh/pkg/server/dns.go]: please add a blank line at the end of this file

File: edgemesh/pkg/server/dns.go (hunk 1 above)
Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: ok，I will fix that

File: edgemesh/pkg/server/dns.go (hunk 2 above)
Comment (kevin-wangzefeng) [reply edgemesh/pkg/server/dns.go]: s/以太网/ethernet/

Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: ok，I will modify this problem

**Pull request merged**